docs/secrets/Secrets.md

```diff
@@ -1,98 +1,217 @@
+__:warning: Warning:__ Any code that gets executed by a pipeline AND has access to the Kubernetes service account token can read all image pull secrets! This is especially important to consider if untrusted code may get executed, e.g. a pipeline processing pull requests from untrusted users.
 
-As first step we only plan to support our predefined Jenkinsfile. But in future teams will be able to use custom pipelines, either from the project sources or another custom git repo.
+- Any code running in the Jenkinsfile Runner container, because this container has the service account token mounted.
+- Any code running in additional containers that have the service account token mounted.
 
-- **Centrally defined Jenkinsfile**: The credentials have to be provided by us.
-- **Custom Jenkinsfile**: Credentials have to be provided by customer
-    - **In Sources**: Same as [Project Source Fetch Secrets]() (see below)
-    - **Separate Repo**: Separate Credentials
+To prevent access to secrets, untrusted code must be executed in containers where the service account token will not be supplied to (mounting of service account token disabled via pod spec and token not passed into the container in any other way).
 
-**Personas:** `CloudCi` & `Customer`
 
+## Other Secrets
 
-### Source Project Secret
+### Log Storage in ElasticSearch
 
-Once the pipeline is running it needs to sync the project sources. This secret might also require write permissions to the repository, e.g. to create a tag, send status feedback, push an updated version commit.
+The log output of pipeline runs can be send to an Elasticsearch server.
+The credentials to authenticate at the Elasticsearch server must be provided by the Steward client.
 
-The required credentials can be provided by customers as [General Jenkins Secrets]() (see below).
+See documenation page [Sending Pipeline Logs to Elasticsearch](../pipeline-logs-elasticsearch/README.md).
 
-**Personas:** `Customer`
+__TODO:__ How to configure credentials for Elasticseach logging
 
 
-## General Jenkins Secrets
+## Links
 
-Customers can define secrets which will be made available to the Jenkins pipeline as regular `Jenkins Credentials` via the `K8s Credential Provider` plugin. Those credentials can be used inside the pipeline, e.g. to deploy to Cloud Foundry, but also to fetch sources from a specific Git repository, etc.
+- Kubernetes Secrets:
+    - the concept of [Secrets][k8s_docs_secrets] (K8s docs)
+    - the definition of secret types in the [source code][k8s_secret_types_src]
+    - [Pull an Image from a Private Registry][k8s_docs-pull_image_private_registry] (K8s docs)
+    - [Add ImagePullSecrets to a service account][k8s_docs_add_imagepullsecrets_to_service_account] (K8s docs)
+    - [Distribute Credentials Securely Using Secrets][k8s_docs_distribute_credentials_secure] (K8s docs)
 
-**Personas:** `CloudCi` & `Customer`
+</p>
```

Comment:
Where is the starting `<p>`? Or did you mean `<p/>`

Reply:
Fixed.